From ebcc8caec2e0f1193860e4052c55a77d62bf6272 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Thu, 15 Feb 2018 15:12:57 -0600
Subject: [PATCH 2725/2748] MAPREDUCE-7052.
 TestFixedLengthInputFormat#testFormatCompressedIn
 is flaky. Contributed by Peter Bacsko

(cherry picked from commit a53d62ab26e170a0338f93e228718da52e9196e4)

Change-Id: I816e595e53065e963614531dccc3902510c83b0b
---
 .../hadoop/mapred/TestFixedLengthInputFormat.java  |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestFixedLengthInputFormat.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestFixedLengthInputFormat.java
index 8013feb..6281e71 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestFixedLengthInputFormat.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestFixedLengthInputFormat.java
@@ -300,7 +300,7 @@ private void runRandomTests(CompressionCodec codec) throws IOException {
       if (i > 0) {
         if (i == (MAX_TESTS-1)) {
           // Test a split size that is less than record len
-          numSplits = (int)(fileSize/Math.floor(recordLength/2));
+          numSplits = (int)(fileSize/ Math.max(1, Math.floor(recordLength/2)));
         } else {
           if (MAX_TESTS % i == 0) {
             // Let us create a split size that is forced to be 
-- 
1.7.9.5

